omnigibson/transition_rules.py

```diff
@@ -277,37 +278,42 @@ def transition(self, individual_objects, group_objects):
         t_results = TransitionResults()
 
         # Load object parts.
-        for _, part_idx in enumerate(sliced_obj.metadata["object_parts"]):
-            # List of dicts gets replaced by {'0':dict, '1':dict, ...}.
-            part = sliced_obj.metadata["object_parts"][part_idx]
-            part_category = part["category"]
-            part_model = part["model"]
+        for part_idx, part in sliced_obj.metadata["object_parts"].items():
+            # List of dicts gets replaced by {'0':dict, '1':dict, ...}
+
+            # Get bounding box info
+            part_bb_pos = np.array(part["bb_pos"])
+            part_bb_orn = np.array(part["bb_orn"])
+
+            # Determine the relative scale to apply to the object part from the original object
+            # Note that proper (rotated) scaling can only be applied when the relative orientation of
+            # the object part is a multiple of 90 degrees wrt the parent object, so we assert that here
+            # Check by making sure the real component of rotation is close to 0, 0.707, or 1
+            assert np.any(np.isclose(np.ones(3) * part_bb_orn[-1], np.array([1.0, 0.7071, 0.0]), atol=1e-3)), \
+                "Sliceable objects should only have relative object part orientations that are factors of 90 degrees!"
+
             # Scale the offset accordingly.
-            part_pos = part["pos"] * sliced_obj.scale
-            part_orn = part["orn"]
-            part_obj_name = f"{sliced_obj.name}_part_{part_idx}"
-            model_root_path = f"{og_dataset_path}/objects/{part_category}/{part_model}"
-            usd_path = f"{model_root_path}/usd/{part_model}.usd"
+            scale = np.abs(T.quat2mat(part_bb_orn) @ sliced_obj.scale)
 
-            # Calculate global part pose.
-            part_pos = np.array(part_pos) + pos
-            part_orn = T.quat_multiply(np.array(part_orn), orn)
+            # Calculate global part bounding box pose.
+            part_bb_pos = pos + T.quat2mat(orn) @ (part_bb_pos * scale)
+            part_bb_orn = T.quat_multiply(orn, part_bb_orn)
 
-            # Circular import.
+            # Avoid circular imports
             from omnigibson.objects.dataset_object import DatasetObject
 
+            part_obj_name = f"{sliced_obj.name}_part_{part_idx}"
             part_obj = DatasetObject(
                 prim_path=f"/World/{part_obj_name}",
-                usd_path=usd_path,
-                category=part_category,
                 name=part_obj_name,
-                scale=sliced_obj.scale,
-                abilities={}
+                category=part["category"],
+                model=part["model"],
+                bounding_box=part["bb_size"] * scale,
```

Comment:
Maybe add a bit more comments regarding what the scale of the child object should be, i.e. it should actually be `part["bb_size"] * scale / self.native_bbox`.

Reply:
done!